configure

```diff
@@ -81,36 +81,92 @@ int main() {
 }
 EOF
 
-# First, try R CMD SHLIB to see if R can already compile
-# things using OpenMP without any extra help from data.table
-"${R_HOME}/bin/R" CMD SHLIB test-omp.c >/dev/null 2>&1 || R_NO_OPENMP=1
-
-if [ "$R_NO_OPENMP" = "1" ]; then
-  # Compilation failed -- try forcing -fopenmp instead.
-  R_NO_OPENMP=0
-  "${CC}" "${CFLAGS}" -fopenmp test-omp.c || R_NO_OPENMP=1
-  # TODO: and then nothing seems to be done with this outcome
-else
-  echo "R CMD SHLIB supports OpenMP without any extra hint"
-fi
+detect_openmp () {
+
+  if [ "$(uname)" = "Linux" ]; then
+
+    printf "%s" "* checking if R installation supports OpenMP without any extra hints... "
+    if "${R_HOME}/bin/R" CMD SHLIB test-omp.c > /dev/null 2>&1; then
+      echo "yes"
+      export R_OPENMP_ENABLED=1
+      return
+    else
+      echo "no"
+    fi
+
 
-# Clean up.
-rm -f test-omp.* a.out
+    printf "%s" "* checking if R installation supports openmp with \"-fopenmp\" flag... "
+    if ${CC} ${CFLAGS} -fopenmp test-omp.c > /dev/null 2>&1; then
```

Comment:
As a side note, note that we intentionally do not quote variable expansions here, since `${CC}` could encode both a compiler name and its flags, e.g. `clang -arch arm64` on macOS.